kafka-product-Events/src/main/java/com/timzowen/kafka/service/impl/ProductServiceImpl.java

```diff
@@ -25,23 +25,25 @@ public ProductServiceImpl(KafkaTemplate<String, ProductCreatedEvent> kafkaTempla
 
     // Step 2 ---> Product event
     @Override
-    public String createProduct(Product product) {
+    public String createProduct(Product product) throws Exception{
+
         String productId = UUID.randomUUID().toString(); // unique identifier for product
+
         // TODO ---> Persist data in db.
+
         ProductCreatedEvent productCreatedEvent =
                 new ProductCreatedEvent(productId, product.getTitle(), product.getPrice(),product.getQuantity());
 
-        CompletableFuture<SendResult<String, ProductCreatedEvent>> future =  kafkaTemplate.send("product-created-events-topic",productId,productCreatedEvent);
-        // listen to completion
-        future.whenComplete((result,exception) -> {
-           if (exception != null){
-               LOGGER.error("********* Failed to send message{}", exception.getMessage());
-           }else {
-               LOGGER.info("********* message sent successfully{}",result.getRecordMetadata());
-           }
-        });
+        LOGGER.info("******** Before sending to kafka topic");
+
+        SendResult<String, ProductCreatedEvent> result =  kafkaTemplate.send("product-created-events-topic",productId,productCreatedEvent).get();
+
+        LOGGER.info("partition: " + result.getRecordMetadata().partition());
```

Comment:
instead fof repeating this , combine this values

LOGGER.info("Sent to topic={} partition={} offset={}",
    result.getRecordMetadata().topic(),
    result.getRecordMetadata().partition(),
    result.getRecordMetadata().offset());

Reply:
approved